File: test/controllers/about_controller_test.rb
require_relative '../test_config'

class AboutControllerTest < MyRecordTest
  include ControllerTesting

  def app
    AboutController
  end

  def test_index_page_is_available
    assert_ok get '/'
  end

  def test_composers_page_is_available
    assert_ok get '/composers'
  end

  def test_search_and_purchase_page_is_available
    assert_ok get '/search_and_purchase'
  end

  def test_licencing_and_copyright_page_is_available
    assert_ok get '/licencing_and_copyright'
  end

  def test_refunds_page_is_available
    assert_ok get '/refunds'
  end

  def test_privacy_page_is_available
    assert_ok get '/privacy'
  end

  def test_terms_and_conditions_page_is_available
    assert_ok get '/terms_and_conditions'
  end

  def test_contact_page_is_available
    assert_ok get '/contact'
  end

  def test_sitemap_is_available
    assert_ok get '/sitemap'
  end

end

File: apps/web/test/web/public/about_controller_test.exs
defmodule UM.Web.AboutControllerTest do
  use ExUnit.Case
  import Raxx.Request

  test "test index page is available" do
    request = get("/")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test composers page is available" do
    request = get("/composers")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test search and purchase page is available" do
    request = get("/search_and_purchase")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test licencing and copyright page is available" do
    request = get("/licencing_and_copyright")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test refunds page is available" do
    request = get("/refunds")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test privacy page is available" do
    request = get("/privacy")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test terms and conditions page is available" do
    request = get("/terms_and_conditions")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test contact page is available" do
    request = get("/contact")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

  test "test sitemap is available" do
    request = get("/sitemap")
    response = UM.Web.AboutController.handle_request(request, [])
    assert 200 == response.status
  end

end
